File: Ticket.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import javafx.scene.image.ImageView;

/**
 *
 * @author oussama
 */
public class Ticket {

    public static void refresh() {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    } 

     
    private int ID;
    String Heurs,Date,Prix,NbPlace,Salle,Film,Promo;
 
    public Ticket(String Prix) {
        this.Prix=Prix;
     }
 

    
    public static int Pdp;
    public static String dates;

    public static String getDates() {
        return dates;
    }

    public static void setDates(String dates) {
        Ticket.dates = dates;
    }

    public static int getPdp() {
        return Pdp;
    }

    public static void setPdp(int Pdp) {
        Ticket.Pdp = Pdp;
    }

    public Ticket() {
       
    }

    public Ticket(String Prix, String NbPlace, String Salle, String Heurs, String Date, String Film, String Promo) {
        this.Prix = Prix;
        this.NbPlace = NbPlace;
        this.Salle = Salle;
        this.Film=Film;
        this.Promo=Promo;
        this.Heurs = Heurs;
        this.Date = Date;
    }

    public Ticket(int ID, String Prix, String NbPlace, String Salle, String Heurs, String Date,  String Film,String Promo) {
        this.ID=ID;
        this.Prix = Prix;
        this.NbPlace = NbPlace;
        this.Salle = Salle;
        this.Film=Film;
        this.Promo=Promo;
        this.Heurs = Heurs;
        this.Date = Date;    }

     
    
 
  
 

    public String getHeurs() {
        return Heurs;
    }

    public void setHeurs(String Heurs) {
        this.Heurs = Heurs;
    }

    public String getDate() {
        return Date;
    }

    public void setDate(String Date) {
        this.Date = Date;
    }

  
    public int getID() {
        return ID;
    }

    public void setID(int ID) {
        this.ID = ID;
    }
public String getPrix() {
        return Prix;
    }

    public void setPrix(String Prix) {
        this.Prix = Prix;
    }

    public String getNbPlace() {
        return NbPlace;
    }

    public void setNbPlace(String NbPlace) {
        this.NbPlace = NbPlace;
    }

    public String getSalle() {
        return Salle;
    }

    public void setSalle(String Salle) {
        this.Salle = Salle;
    }

    public String getFilm() {
        return Film;
    }

    public void setFilm(String Film) {
        this.Film = Film;
    }

    public String getPromo() {
        return Promo;
    }

    public void setPromo(String Promo) {
        this.Promo = Promo;
    }

    @Override
    public String toString() {
        return "Ticket{" + "ID=" + ID + ", Heurs=" + Heurs + ", Date=" + Date + ", Prix=" + Prix + ", NbPlace=" + NbPlace + ", Salle=" + Salle + ", Film=" + Film + ", Promo=" + Promo + '}';
    }

 

 

   
    
  
    public void AddTicket(Ticket t) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

     
    
}

File: Ticket.php
<?php

namespace App\Entity;

use App\Repository\TicketRepository;
use  Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\Promotion;

/**
 * @ORM\Entity(repositoryClass=TicketRepository::class)
 */
class Ticket
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $PrixTicket;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $NbPlace;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $Salle;

    /**
     * @ORM\Column(type="date")
     * @Assert\NotBlank(message="Ce champs est obligatoire")

     */
    private $dateh;


    /**
     * @ORM\ManyToOne(targetEntity=Promotion::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private $promo;

    /**
     * @ORM\ManyToOne(targetEntity=Film::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private $film;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPrixTicket(): ?string
    {
        return $this->PrixTicket;
    }

    public function setPrixTicket(string $PrixTicket): self
    {
        $this->PrixTicket = $PrixTicket;

        return $this;
    }

    public function getNbPlace(): ?string
    {
        return $this->NbPlace;
    }

    public function setNbPlace(string $NbPlace): self
    {
        $this->NbPlace = $NbPlace;

        return $this;
    }

    public function getSalle(): ?string
    {
        return $this->Salle;
    }

    public function setSalle(string $Salle): self
    {
        $this->Salle = $Salle;

        return $this;
    }


    public function getDateh(): ?\DateTimeInterface
    {
        return $this->dateh;
    }

    public function setDateh(\DateTimeInterface $dateh): self
    {
        $this->dateh = $dateh;

        return $this;
    }

    public function getPromo(): ?Promotion
    {
        return $this->promo;
    }

    public function setPromo(?Promotion $promo): self
    {
        $this->promo = $promo;

        return $this;
    }

    public function getFilm(): ?Film
    {
        return $this->film;
    }

    public function setFilm(?Film $film): self
    {
        $this->film = $film;

        return $this;
    }

    public function __toString()
    {
        return (string) $this->getFilm();
    }

}
